### PR TITLE
EPMRPP-33560. Launch level pie charts implementation.

### PR DESCRIPTION
**Pull request by ArtsiomYeliseyenka**

### CODE REVIEW

File: app/src/pages/inside/launchesPage/launchSuiteGrid/defectStatistics/pieChart/pieChart.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      .forEach((key) => {
+        if (key !== 'total') {
+          const defectType = this.props.projectConfig.subTypes[this.props.type.toUpperCase()];
+          data.columns.push([key, this.props.data[key]]);
+          data.colors[key] = defectType.find(item => item.locator === key).color;
+        }
+      });
+    return data;
+  };
+  chartConfig = {
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/launchSuiteGrid/defectStatistics/pieChart/pieChart.jsx]: Can be a local constant

File: app/src/pages/inside/launchesPage/launchSuiteGrid/executionStatistics/executionStatistics.jsx [hunk 2]
```diff
@@ -7,9 +7,14 @@ const cx = classNames.bind(styles);
 export const ExecutionStatistics = ({ value, title, bold }) => (
   <div className={cx('execution-statistics')}>
     <span className={cx('title')}>{ title.full }</span>
-    <a href="/" className={cx('value', { bold })}>{ value }</a>
+    {
+      !!+value &&
```
Comment (DiscoElevator) [inline app/src/pages/inside/launchesPage/launchSuiteGrid/executionStatistics/executionStatistics.jsx]: `!!Number(value)`

File: app/src/pages/inside/launchesPage/defectTypeTooltip/defectTypeTooltip.jsx [hunk 3]
```diff
@@ -44,17 +45,21 @@ export class DefectTypeTooltip extends Component {
...
-                <div className={cx('circle')} style={{ backgroundColor: item.color }} />
-                { item.longName }
-              </div>
-              <span className={cx('value')}>
-                { this.props.data[item.locator] }
-              </span>
-            </a>
-            ))
+          Object.keys(this.props.data).map((key) => {
+            const defectType = defectConfig.find(item => item.locator === key);
```
[the remark this answers is not in the document]
Comment (ArtsiomYeliseyenka) [reply app/src/pages/inside/launchesPage/defectTypeTooltip/defectTypeTooltip.jsx]: fixed. Added polyfill.

File: app/src/pages/inside/launchesPage/launchSuiteGrid/defectStatistics/pieChart/pieChart.jsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    align: 'right',
+    noArrow: true,
+    desktopOnly: true,
+    verticalOffset: -10,
+  },
+})
+@connect(state => ({
+  projectConfig: projectConfigSelector(state),
+}))
+export class PieChart extends PureComponent {
```
[the remark this answers is not in the document]
Comment (ArtsiomYeliseyenka) [reply app/src/pages/inside/launchesPage/launchSuiteGrid/defectStatistics/pieChart/pieChart.jsx]: fixed

File: app/src/pages/inside/launchesPage/launchSuiteGrid/defectStatistics/pieChart/pieChart.jsx (hunk 1 above)
Comment (ArtsiomYeliseyenka) [reply app/src/pages/inside/launchesPage/launchSuiteGrid/defectStatistics/pieChart/pieChart.jsx]: fixed

File: app/src/pages/inside/launchesPage/launchSuiteGrid/executionStatistics/executionStatistics.jsx (hunk 2 above)
Comment (ArtsiomYeliseyenka) [reply app/src/pages/inside/launchesPage/launchSuiteGrid/executionStatistics/executionStatistics.jsx]: fixed

File: app/src/pages/inside/launchesPage/launchSuiteGrid/defectStatistics/pieChart/pieChart.jsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      type: 'donut',
+      order: null,
+      colors: {},
+    };
+    Object.keys(this.props.data)
+      .forEach((key) => {
+        if (key !== 'total') {
+          const defectType = this.props.projectConfig.subTypes[this.props.type.toUpperCase()];
+          data.columns.push([key, this.props.data[key]]);
+          data.colors[key] = defectType.find(item => item.locator === key).color;
```
[the remark this answers is not in the document]
Comment (ArtsiomYeliseyenka) [reply app/src/pages/inside/launchesPage/launchSuiteGrid/defectStatistics/pieChart/pieChart.jsx]: done

**Pull request merged**